### DISCUSSION TITLE
How to turn on or off gpu

### DISCUSSION THREAD

**Discussion by GoldDRoge** [pyannote/pyannote-audio#1095] [Development]
I want to control when to turn on or off gpu but i cant find how to do that. Can u help me

**Comment by hbredin** [comment]
pyannote pipelines will always use GPU when there is one available.

You could set [`CUDA_VISIBLE_DEVICES` environment variable](https://developer.nvidia.com/blog/cuda-pro-tip-control-gpu-visibility-cuda_visible_devices/) to  hide GPUs:

```python
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""

from pyannote.audio import Pipeline
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization")
pipeline("audio.wav") # will not use GPU
```

**Comment by hbredin** [comment]
[This PR](https://github.com/pyannote/pyannote-audio/pull/1204) adds a new `pipeline.to(device)` API. Would you mind testing it?

**Reply by GoldDRoge** [reply]
thanks for your help. its work
